components/api-controller/pkg/controller/v1alpha2/crd.go

```diff
@@ -100,10 +100,14 @@ func Crd() *k8sApiExtensions.CustomResourceDefinition {
 }
 
 const (
-	hostnamePattern = `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`
+	hostnamePatternFormat = `^([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])(.%s)*$`
```

Comment:
I'm not a regexp master, but I think there is a lot of invalid hostnames that will pass the validation. Try the following:
- `host/kyma.local`
- `host.kyma*local`
- `host.kyma.local.kyma.local.kyma.local`
All of them will be valid hostnames for domain kyma.local.
To avoid it please:
- replace `*` with `?` for second group
- replace `.` with `\.` in second group
- escape `.` in domain name

Reply:
You're absolutely right.